checker/jdk/signedness/src/java/lang/Integer.java

```diff
@@ -0,0 +1,1602 @@
+     * or larger than {@code Character.MAX_RADIX}, then the radix
+     * {@code 10} is used instead.
+     *
+     * <p>Note that since the first argument is treated as an unsigned
+     * value, no leading sign character is printed.
+     *
+     * <p>If the magnitude is zero, it is represented by a single zero
+     * character {@code '0'} ({@code '\u005Cu0030'}); otherwise,
+     * the first character of the representation of the magnitude will
+     * not be the zero character.
+     *
+     * <p>The behavior of radixes and the characters used as digits
+     * are the same as {@link #toString(int, int) toString}.
+     *
+     * @param   i       an integer to be converted to an unsigned string.
+     * @param   radix   the radix to use in the string representation.
+     * @return  an unsigned string representation of the argument in the specified radix.
+     * @see     #toString(int, int)
+     * @since 1.8
+     */
+    public static String toUnsignedString(@Unsigned int i, int radix) {
+        return Long.toUnsignedString(toUnsignedLong(i), radix);
+    }
+
+    /**
+     * Returns a string representation of the integer argument as an
+     * unsigned integer in base&nbsp;16.
+     *
+     * <p>The unsigned integer value is the argument plus 2<sup>32</sup>
+     * if the argument is negative; otherwise, it is equal to the
+     * argument.  This value is converted to a string of ASCII digits
+     * in hexadecimal (base&nbsp;16) with no extra leading
+     * {@code 0}s.
+     *
+     * <p>The value of the argument can be recovered from the returned
+     * string {@code s} by calling {@link
+     * Integer#parseUnsignedInt(String, int)
+     * Integer.parseUnsignedInt(s, 16)}.
+     *
+     * <p>If the unsigned magnitude is zero, it is represented by a
+     * single zero character {@code '0'} ({@code '\u005Cu0030'});
+     * otherwise, the first character of the representation of the
+     * unsigned magnitude will not be the zero character. The
+     * following characters are used as hexadecimal digits:
+     *
+     * <blockquote>
+     *  {@code 0123456789abcdef}
+     * </blockquote>
+     *
+     * These are the characters {@code '\u005Cu0030'} through
+     * {@code '\u005Cu0039'} and {@code '\u005Cu0061'} through
+     * {@code '\u005Cu0066'}. If uppercase letters are
+     * desired, the {@link java.lang.String#toUpperCase()} method may
+     * be called on the result:
+     *
+     * <blockquote>
+     *  {@code Integer.toHexString(n).toUpperCase()}
+     * </blockquote>
+     *
+     * @param   i   an integer to be converted to a string.
+     * @return  the string representation of the unsigned integer value
+     *          represented by the argument in hexadecimal (base&nbsp;16).
+     * @see #parseUnsignedInt(String, int)
+     * @see #toUnsignedString(int, int)
+     * @since   JDK1.0.2
+     */
+    public static String toHexString(@UnknownSignedness int i) {
```

Comment:
toUnsignedString0 takes an unsigned string (and passes it to a method documented as taking an unsigned string), so I think toHexString should too.
The same goes for to toOctalString and toBinaryString.

Reply:
Fixed this!